src/lib/storage/value_segment.cpp

```diff
@@ -7,67 +7,80 @@ namespace opossum {
 
 template <typename T>
 ValueSegment<T>::ValueSegment(bool nullable) {
-  // Implementation goes here
+  _value_vector = std::vector<T>(0);
+  _is_null_vector = std::vector<bool>(0);
+  _segment_is_nullable = nullable;
 }
 
 template <typename T>
 AllTypeVariant ValueSegment<T>::operator[](const ChunkOffset chunk_offset) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  if (_is_null_vector[chunk_offset]) {
+    return NULL_VALUE;
+  }
+  return _value_vector[chunk_offset];
```

Comment:
Consider using `is_null()` to avoid code duplication.
Also: `operator[]()` does not do bounds checking, consider using `at()`.